src/csharp/Microsoft.Spark/Utils/CommandSerDe.cs

```diff
@@ -212,7 +225,42 @@ private sealed class UdfWrapperData
 
             foreach (UdfSerDe.FieldData field in fields)
             {
-                SerializeUdfs((Delegate)field.ValueData.Value, curNode, udfWrapperNodes, udfs);
+                SerializeUdfs(
+                    (Delegate)field.ValueData.Value,
+                    curNode,
+                    udfWrapperNodes,
+                    udfs,
+                    udfRefAssemblies);
+            }
+        }
+
+        private static void SerializeReferencedAssemblies(
+            AssemblyName[] referencedAssemblies,
+            Dictionary<string, string> udfRefAssemblies)
+        {
+            foreach (AssemblyName assemblyName in referencedAssemblies)
+            {
+                if (!udfRefAssemblies.ContainsKey(assemblyName.FullName))
+                {
+                    try
+                    {
+                        Assembly asm = Assembly.Load(assemblyName.FullName);
+                        udfRefAssemblies.Add(assemblyName.FullName, asm.ManifestModule.Name);
```

Comment:
Is `ManifestModule.Name` guaranteed to be the name of the file on disk? I would have expected this to use `asm.Location`.

Reply:
I'm not sure how guaranteed it is, but from the description it seems to be similar to asm.Location except with the path removed.

Reply:
Since we're not sure, I'll use asm.Location just in case.